src/actions/AuctionActions.js

```diff
@@ -103,11 +125,46 @@ export function toggleAuctionDetail(auctionId) {
         }
     }
 
-
 }
 
 export function clearAuctionDetail() {
     return {
         type: CLEAR_AUCTION_DETAIL
     }
+}
+
+function buildConfirmUpdateObj(auctions) {
+    let updateObj = {};
+    auctions.forEach( auction => {
+        let objStr = `${auction.id}/winningBids`;
+        updateObj[objStr] = combineWinningBids(auction);
+    })
+    return updateObj;
+}
+
+function combineWinningBids(auction) {
+    
+    let winningBidsObj = auction.winningBids || {};
+    let winningBids = Object.keys(auction.bids).filter( bid => auction.bids[bid].checked );
+    
+    if (!auction.winningBids) {
+        winningBids.forEach( bidId => {
+            delete auction.bids[bidId].checked;
+            delete auction.bids[bidId].winner;
+            winningBidsObj[bidId] = auction.bids[bidId]
+        })
+    } else {
+    
+        let newWinningBids = winningBids.filter( bidId => !winningBids[bidId] );
+        winningBidsObj = assign(winningBidsObj, auction.winningBids);
+            
+        newWinningBids.forEach( bidId => {
+            delete auction.bids[bidId].checked;
+            delete auction.bids[bidId].winner;
+            winningBidsObj[bidId] = auction.bids[bidId]
+        })
+
+    }
+
+    return winningBidsObj;
```

Comment:
I agree, the total sum logic can be refactored and we'll replace the total of all auctions.  After core functionality is merged into Dev I"ll continue improving this feature.